src/uct/tcp/tcp_ep.c

```diff
@@ -615,7 +615,8 @@ static inline ssize_t uct_tcp_ep_sendv(uct_tcp_ep_t *ep)
     size_t sent_length;
     ucs_status_t status;
 
-    ucs_assertv(ep->tx.offset < ep->tx.length, "ep=%p", ep);
+    ucs_assertv((ep->tx.offset < ep->tx.length) &&
```

Comment:
can offset be more than length if no iovs?

Reply:
a, you are checking that it is not 0-byte send?

Reply:
yes, TCP has to not have 0-byte `send()`s or `sendv()`s